### DISCUSSION TITLE
Cannot load the graph: cluster [xxx] is not found or is not accessible for Kiali

### DISCUSSION THREAD

**Discussion by wangshiyang** [kiali/kiali#7840] [Q&A] [answered]
Describe the bug:

Cannot load the graph: cluster [xxx] is not found or is not accessible for Kiali

Expected Behavior:

1. Set up two Kubernetes clusters.
2. Install Istio for a multi-master cluster. (https://istio.io/latest/docs/setup/install/multicluster/multi-primary_multi-network/)
3. Install Istio in cluster1 -> samples -> addons.

Environment:

Kiali version: 1.89.7
- **Istio version: 1.22.3**
- **Kubernetes version:1.30.3** 
- 
<img width="1411" alt="image" src="https://github.com/user-attachments/assets/ee464a7c-ec94-423b-a51f-8bb066181270">

**Comment by jshaughn** [comment]
@wangshiyang This sort of error usually supplies a stack trace, can you post it here?

**Comment by jshaughn** [comment]
@wangshiyang I don't think the reproduction steps are detailed enough to reproduce this issue.  If you can post the stack trace we may be able to resolve the problem, otherwise I will have to close this issue.

**Comment by wangshiyang** [comment]
> @wangshiyang I don't think the reproduction steps are detailed enough to reproduce this issue. If you can post the stack trace we may be able to resolve the problem, otherwise I will have to close this issue.

`cluster [cluster2] is not found or is not accessible for Kiali
Additional Detail:
goroutine 4097561 [running]:
runtime/debug.Stack()
	/opt/hostedtoolcache/go/1.22.1/x64/src/runtime/debug/stack.go:24 +0x5e
github.com/kiali/kiali/handlers.handlePanic({0x26e5f70, 0xc00102c138})
	/home/runner/work/kiali/kiali/handlers/graph.go:86 +0x1ad
panic({0x1e65e80?, 0xc0000656a0?})
	/opt/hostedtoolcache/go/1.22.1/x64/src/runtime/panic.go:770 +0x132
github.com/kiali/kiali/graph.CheckError(...)
	/home/runner/work/kiali/kiali/graph/util.go:38
github.com/kiali/kiali/graph/telemetry/istio/appender.addBadging(0xc001566090, 0xc001566030, 0xc000df6030)
	/home/runner/work/kiali/kiali/graph/telemetry/istio/appender/istio_details.go:63 +0x445
github.com/kiali/kiali/graph/telemetry/istio/appender.IstioAppender.AppendGraph({0x236ed5c?}, 0xc001566090, 0xc001566030, 0xc000df6030)
	/home/runner/work/kiali/kiali/graph/telemetry/istio/appender/istio_details.go:48 +0x73
github.com/kiali/kiali/graph/telemetry/istio.BuildNamespacesTrafficMap({_, _}, {0xc00047dc20, {0x0, {0xc000f5f1a0, 0x6, 0x6}}, 0x0, 0x1, 0xc00047d920, ...}, ...)
	/home/runner/work/kiali/kiali/graph/telemetry/istio/istio.go:80 +0x674
github.com/kiali/kiali/graph/api.graphNamespacesIstio({_, _}, _, _, {{0x2374a39, 0x9}, {0x236ed5c, 0x5}, {{0xc00146a07b, 0x15}, ...}, ...})
	/home/runner/work/kiali/kiali/graph/api/api.go:52 +0xf0
github.com/kiali/kiali/graph/api.GraphNamespaces({_, _}, _, {{0x2374a39, 0x9}, {0x236ed5c, 0x5}, {{0xc00146a07b, 0x15}, {0x1a3185c5000, ...}}, ...})
	/home/runner/work/kiali/kiali/graph/api/api.go:33 +0x2f3
github.com/kiali/kiali/handlers.GraphNamespaces({0x26e5f70, 0xc00102c138}, 0xc001272a20)
	/home/runner/work/kiali/kiali/handlers/graph.go:51 +0x165
net/http.HandlerFunc.ServeHTTP(0x237ebc7?, {0x26e5f70?, 0xc00102c138?}, 0x120?)
	/opt/hostedtoolcache/go/1.22.1/x64/src/net/http/server.go:2166 +0x29
github.com/kiali/kiali/routing.NewRouter.metricHandler.func6({0x26e5ee0, 0xc000f4c0e0}, 0xc001272a20)
	/home/runner/work/kiali/kiali/routing/router.go:246 +0x11f
net/http.HandlerFunc.ServeHTTP(0xc001272900?, {0x26e5ee0?, 0xc000f4c0e0?}, 0xc0001c0900?)
	/opt/hostedtoolcache/go/1.22.1/x64/src/net/http/server.go:2166 +0x29
github.com/kiali/kiali/routing.NewRouter.AuthenticationHandler.Handle.func7({0x26e5ee0, 0xc000f4c0e0}, 0xc001272900)
	/home/runner/work/kiali/kiali/handlers/authentication.go:70 +0x31f
net/http.HandlerFunc.ServeHTTP(0x26f3480?, {0x26e5ee0?, 0xc000f4c0e0?}, 0x26be3b0?)
	/opt/hostedtoolcache/go/1.22.1/x64/src/net/http/server.go:2166 +0x29
github.com/kiali/kiali/server.plainHttpMiddleware.func1({0x26e5ee0?, 0xc000f4c0e0?}, 0xc00047d800?)
	/home/runner/work/kiali/kiali/server/server.go:185 +0x5c
net/http.HandlerFunc.ServeHTTP(0xc0012727e0?, {0x26e5ee0?, 0xc000f4c0e0?}, 0x41a298?)
	/opt/hostedtoolcache/go/1.22.1/x64/src/net/http/server.go:2166 +0x29
github.com/gorilla/mux.(*Router).ServeHTTP(0xc000954fc0, {0x26e5ee0, 0xc000f4c0e0}, 0xc0012726c0)
	/home/runner/go/pkg/mod/github.com/gorilla/mux@v1.8.1/mux.go:212 +0x1e2
github.com/NYTimes/gziphandler.GzipHandlerWithOpts.func1.1({0x26df4c0, 0xc0000d9340}, 0xc0012726c0)
	/home/runner/go/pkg/mod/github.com/!n!y!times/gziphandler@v1.1.1/gzip.go:336 +0x27e
net/http.HandlerFunc.ServeHTTP(0xc0008ad520?, {0x26df4c0?, 0xc0000d9340?}, 0x6fc41a?)
	/opt/hostedtoolcache/go/1.22.1/x64/src/net/http/server.go:2166 +0x29
net/http.(*ServeMux).ServeHTTP(0x471639?, {0x26df4c0, 0xc0000d9340}, 0xc0012726c0)
	/opt/hostedtoolcache/go/1.22.1/x64/src/net/http/server.go:2683 +0x1ad
net/http.serverHandler.ServeHTTP({0xc00047d440?}, {0x26df4c0?, 0xc0000d9340?}, 0x6?)
	/opt/hostedtoolcache/go/1.22.1/x64/src/net/http/server.go:3137 +0x8e
net/http.(*conn).serve(0xc000824120, {0x26f3480, 0xc0006fa150})
	/opt/hostedtoolcache/go/1.22.1/x64/src/net/http/server.go:2039 +0x5e8
created by net/http.(*Server).Serve in goroutine 446
	/opt/hostedtoolcache/go/1.22.1/x64/src/net/http/server.go:3285 +0x4b4`

**Comment by jshaughn** [comment]
Thank you, @wangshiyang! I'll see if I can spot the problem.

**Comment by jshaughn** [comment]
Hi @wangshiyang , Are you absolutely sure you are using Kiali version: 1.89.7?  Using the About Box in the UI masthead,  please verify that both the UI and Server are the expected version.  The reason I ask is that a fix for this exact problem (same stack trace) was in v1.89.3 (https://github.com/kiali/kiali/issues/7672).

Please let me know, I don't currently see another problem in the code.

**Comment by wangshiyang** [comment, marked as answer]
It was a version issue; after the update, it works properly now.

**Reply by jshaughn** [reply]
That is good to hear because I couldn't find a problem!  Thanks for checking.
